### PR TITLE
WAF: Fix invalid headers

### PR DESCRIPTION
**Pull request by jwidavid**
This PR adds a sanitation method to the runtime class. The output that is generated when a request is blocked will now be sanitized before being logged. Further, we will now just return the rule ID in the header response for blocked requests.

Asana [#1201890219548759](https://app.asana.com/0/1201069996155224/1201890219548759)

#### Changes proposed in this Pull Request:

* Adds sanitation_output method to the `class-waf-runtime` class.
* The `class-waf-runtime` block method now uses the sanitize_output method before logging the reason for blocking.
* The block method now only returns the rule ID rather than the reason in the response header.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Update your Jurassic Tube test site's rules.php file with the latest from https://github.com/Automattic/jetpack-waf-rules/blob/trunk/waf-rules/rules.php
* Run `curl -D - -s --url "https://<your-sub-domain>.jurassic.tube/xss" --data "a=<svg/onload%0c=alert%601%60>"`
* You should see a 500 error response with a 403 error log
* Update your site with this PR (ensure rules.php is also re-updated)
* Running the curl no longer generates a 500 error
* Run `curl -D - -s --url "https://<your-sub-domain>.jurassic.tube?jetpack-waf-always-block"`

**Expectation:** The response should now say "403 - rule 1337", the reason should be printed below the response header.

### CODE REVIEW

File: projects/packages/waf/src/class-waf-runtime.php [hunk 1]
```diff
@@ -240,15 +240,17 @@ public function match_targets( $transforms, $targets, $match_operator, $match_va
...
 
 		$this->write_blocklog( $rule_id, $reason );
 		error_log( "Jetpack WAF Blocked Request\t$action\t$rule_id\t$status_code\t$reason" );
-		header( "X-JetpackWAF-Blocked: $status_code $reason" );
+		header( "X-JetpackWAF-Blocked: $status_code - rule $rule_id" );
 		if ( defined( 'JETPACK_WAF_MODE' ) && 'normal' === JETPACK_WAF_MODE ) {
 			$protocol = isset( $_SERVER['SERVER_PROTOCOL'] ) ? wp_unslash( $_SERVER['SERVER_PROTOCOL'] ) : 'HTTP';
 			header( $protocol . ' 403 Forbidden', true, $status_code );
-			die( "rule $rule_id" );
+			die( "rule $rule_id - reason $reason" );
```
Comment (nateweller) [inline projects/packages/waf/src/class-waf-runtime.php:253]: Not major but if there is no reason available, I believe we'll have a redundant output here of `rule $rule_id - reason rule $rule_id`.

**Comment by nateweller** [comment]
Tests perfectly. Code looks great too. One little note about the redundant `die()` message but everything else looks good to me! :+1:

Output from my tests:

```
curl -D - -s --url "https://njweller.jurassic.tube/xss" --data "a=<svg/onload%0c=alert%601%60>"

HTTP/1.1 403 Forbidden
Server: nginx/1.14.0 (Ubuntu)
Date: Mon, 25 Apr 2022 16:14:15 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 163
Connection: keep-alive
X-JetpackWAF-Blocked: 403 - rule 941160

rule 941160 - reason NoScript XSS InjectionChecker: HTML Injection Matched Data: &amp;lt;svg\/onload\b= found within args:a: &amp;lt;svg\/onload\b=alert`1`&amp;gt;        
```

```
curl -D - -s --url "https://njweller.jurassic.tube?jetpack-waf-always-block"  
                 
HTTP/1.1 403 Forbidden
Server: nginx/1.14.0 (Ubuntu)
Date: Mon, 25 Apr 2022 16:15:47 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 59
Connection: keep-alive
X-JetpackWAF-Blocked: 403 - rule 1337

rule 1337 - reason Intentionally blocking this test-request
```

**Review by nateweller — DISMISSED** [review]

File: projects/packages/waf/src/class-waf-runtime.php (hunk 1 above)
Comment (jwidavid) [reply projects/packages/waf/src/class-waf-runtime.php:253]: @nateweller Which part is redundant? This should just be adding the reason to the die which was information we weren't displaying previously. The rule ID is already printed out to console (so technically that shows up twice), but I didn't add that. Plus, the die is the only thing that will be seen when it is executed on a site as opposed to running from a terminal curl I believe?

Comment (nateweller) [reply projects/packages/waf/src/class-waf-runtime.php:253]: Just the minor fact that the `die()` string output will repeat itself by including the "rule $rule_id" string as both the prefix and the reason:

"rule ABC123 - reason rule ABC123"

Due to `$reason` being set as "rule ABC123" when no more detailed reason is available.

**Review by nateweller — DISMISSED** [review]

File: projects/packages/waf/src/class-waf-runtime.php (hunk 1 above)
Comment (jwidavid) [reply projects/packages/waf/src/class-waf-runtime.php:253]: Ah - right, though that's only if there isn't a reason. There should however always be a reason provided. Normal expectation would be "rule xx - reason blah blah blah" which is just a little more helpful for troubleshooting. The die as it exists won't be a part of our final product, so this is more for testing.

Comment (nateweller) [reply projects/packages/waf/src/class-waf-runtime.php:253]: Sounds good to me!

**Review by nateweller — APPROVED** [review]

**Pull request merged**